### PR TITLE
[DOCS] Overhaul search template docs

### LINKED ISSUE #72467: [DOCS] Search template tutorial is in reference docs
Search templates are a useful feature when using ES as a search backend. Currently, the search template tutorial is part of the search template API reference docs. However, creating and managing search templates involves several other APIs.

We should consider breaking out the tutorial and adding it to the "Search your data" section. We should also add reference docs for the render search template API.

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Adds a tutorial for search templates.
* Adds reference docs for the render search template API.
* Splits the source files for the search template API and the multi search template API docs.
* Improves parameter documentation for the multi search template API.
* Removes duplicate examples from the search template API, multi search API, and create stored script API docs.

Closes #72467

### Previews
* Search templates tutorial: https://elasticsearch_72583.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-template.html
* Search template API docs: https://elasticsearch_72583.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-template-api.html
* Multi search template API docs: https://elasticsearch_72583.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/multi-search-template.html
* Render search template API docs: https://elasticsearch_72583.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/render-search-template-api.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@elastimachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/search/render-search-template-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+`<template-id>`::
+(Optional, string) ID of the search template to render. If no `source` is
+specified, this or the `id` request body parameter is required.
+
+[[render-search-template-api-request-body]]
+==== {api-request-body-title}
+
+`id`::
+(Required*, string) ID of the search template to render. If no `source` is
```
Comment (cbuescher) [inline docs/reference/search/render-search-template-api.asciidoc:76]: I stumbled across the "*" syntax here, but I saw this is used elsewhere already. Just wanted to let you know I didn't really know what to make of it at first, but when I read both parameter entries it makes sense.

**Review by cbuescher — APPROVED** [review]
LGTM, I didn't compare all examples with what we had before but I think they are mostly moved from the old location. I like the separation of the tutorial and the reference, makes a lot more sense this way.

File: docs/reference/search/search-your-data/search-template.asciidoc [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+<<search-search-api-request-body,search API>>'s request body. `source` also
+supports https://mustache.github.io/[Mustache] variables, typically enclosed in
+double curly brackets: `{{my-var}}`. When you run a templated search, {es}
+replaces these variables with values from `params`.
+
+Search templates must use a `lang` of `mustache`.
+
+[source,console]
+----
+PUT _scripts/my-search-template
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/search/search-your-data/search-template.asciidoc:33]: Thanks. I added a sentence to call out the ID.

**Comment by jrodewig** [comment]
Thanks for the review @cbuescher!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/eql-correctness

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**